pkg/nvlib/info/root.go

```diff
@@ -0,0 +1,81 @@
+/**
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package info
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/NVIDIA/go-nvml/pkg/dl"
+)
+
+type root string
```

Comment:
Can we call this rootPath?

Reply:
The reason for a more generic name than `*Path` is that this type is intended to be used to extract more information about the driver root. There are multiple implementations of it in various projects and I need to consolidate them at some point soon.

The idea is that this struct will at least be able to:
* Return the path to the driver libraries at this root.
* Return the path to the device nodes at this path (or `/`)

I don't know whether the `Path` suffix conveys this.

I'm happy to make the change here now if you feel strongly about it since we don't have the other functionality consolidated yet.